api/src/main/java/org/pmiops/workbench/actionaudit/adapters/ProfileAuditAdapterImpl.kt

```diff
@@ -0,0 +1,116 @@
+package org.pmiops.workbench.actionaudit.adapters
+
+import java.time.Clock
+import java.util.logging.Level
+import java.util.logging.Logger
+import javax.inject.Provider
+import org.pmiops.workbench.actionaudit.ActionAuditEvent
+import org.pmiops.workbench.actionaudit.ActionAuditService
+import org.pmiops.workbench.actionaudit.ActionType
+import org.pmiops.workbench.actionaudit.AgentType
+import org.pmiops.workbench.actionaudit.TargetType
+import org.pmiops.workbench.actionaudit.targetproperties.ProfileTargetProperty
+import org.pmiops.workbench.actionaudit.targetproperties.TargetPropertyExtractor
+import org.pmiops.workbench.db.model.DbUser
+import org.pmiops.workbench.model.Profile
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Service
+
+@Service
+class ProfileAuditAdapterImpl @Autowired
+constructor(
+    private val userProvider: Provider<DbUser>,
+    private val actionAuditService: ActionAuditService,
+    private val clock: Clock,
+    @Qualifier("ACTION_ID") private val actionIdProvider: Provider<String>
+) : ProfileAuditAdapter {
+
+    override fun fireCreateAction(createdProfile: Profile) {
+        try {
+            val propertiesByName: Map<String, String?> = TargetPropertyExtractor
+                    .getPropertyValuesByName(ProfileTargetProperty.values(), createdProfile)
+            val actionId = actionIdProvider.get()
+            // We can't rely on the UserProvider here since the user didn't exist when it
+            // got loaded or whatever.
+            val createEvents = propertiesByName.entries
+                    .map {
```

Comment:
You can still use method references in places where the iterator isn't necessary. Since `it` is a [keyword](https://kotlinlang.org/docs/reference/lambdas.html#it-implicit-name-of-a-single-parameter) I think you'd have to learn to love it, especially when reading third-party Kotlin code and examples. What I do like about it is that whatever you think of how they named it, there's no argument to have over spelling or choice of parameter.

There are cases where names help, for example in long chains of maps. But I think this is just as not just idiomatic, but a core part of the language.